### ISSUE TITLE
eq_op false positive with sub and div

### ISSUE THREAD

**Issue by pgimalac** [rust-lang/rust-clippy#6213]
If you do something like
```Rust
x - x
```
or
```Rust
x / x
```
clippy will throw an eq_op error, while I think it shouldn't.

In my case x is a generic `<E: Sub<Output = E> + Add<Output = E> + Ord>` and I do `x - x > x` to check if x is lower than *zero*.

The same thing could happen with `/` and *one* I guess.

I avoided the error by writing `x > x + x` but I still think it shouldn't be considered an error when you don't know what your type is.

**Comment by llogiq** [comment]
I think that this is a pretty niche thing to do, so using `#[allow(clippy::eq_op)]` should suffice, especially if the lint triggers only on a few isolated functions.

**Comment by pgimalac** [comment]
Oh I didn't know you could ignore some lints like that, that's nice (I usually just run clippy and see where I messed up, it's the first time it was not a real error).

I agree that it probably is not a very common thing to do, and I'm sure this lint is usually useful.

I'm closing the issue, thanks for the answer.
